### PR TITLE
retrieved the current user's details from the API

### PR DESCRIPTION
**Pull request by Nathaniel-Akubuo**
Set up the profile screen view model to retrieve the current user's profile details and display it in the profile view

### CODE REVIEW

File: lib/ui/shared/bottom_sheets/custom_user_bottom_sheet_viewmodel.dart [hunk 1]
```diff
@@ -1,16 +1,31 @@
...
+  final _storage = locator<SharedPreferenceLocalStorage>();
+  final _apiService = locator<HttpApiService>();
+  UserModel? _userModel;
 
-  Future setStatus() async {
-    await _navigationService.navigateTo(Routes.setStatusView);
-  }
+  void setStatus() => _navigationService.navigateTo(Routes.setStatusView);
+
+  void editProfile() => _navigationService.navigateTo(Routes.editProfileView);
```
Comment (isaacadariku) [inline lib/ui/shared/bottom_sheets/custom_user_bottom_sheet_viewmodel.dart:18]: can you rename this method, to be `navigateToEditProfile`

File: lib/ui/shared/bottom_sheets/custom_user_bottom_sheet_viewmodel.dart [hunk 2]
```diff
@@ -1,16 +1,31 @@
...
+class CustomUserBottomSheetViewModel extends FutureViewModel {
   final _navigationService = locator<NavigationService>();
+  final _storage = locator<SharedPreferenceLocalStorage>();
+  final _apiService = locator<HttpApiService>();
+  UserModel? _userModel;
 
-  Future setStatus() async {
-    await _navigationService.navigateTo(Routes.setStatusView);
-  }
+  void setStatus() => _navigationService.navigateTo(Routes.setStatusView);
```
Comment (isaacadariku) [inline lib/ui/shared/bottom_sheets/custom_user_bottom_sheet_viewmodel.dart:16]: Also this should have a proper naming `navigateToSetStatus`

**Review by isaacadariku — DISMISSED** [review]
Great!, Just few changes and we are good to go

**Review by Blazebrain — APPROVED** [review]
It's good to go

**Review by mayor04 — APPROVED** [review]

**Review by Quwaysim — APPROVED** [review]
lgtm

**Pull request merged**